### ISSUE TITLE
`view` is undefined

### ISSUE THREAD

**Issue by code-hunger** [minbrowser/min#610]
I started some refactoring on the tabState and after moving the taskPrototype to its own file, I started getting some bizarre errors.

After I put guards and `throw`s around all errors I got, I discovered the error in the browser console - which was inaccessible before I did this, as the browser was constantly alert()ing errors. 

Here's the commit with the checks that failed: add9b68b127e147d150e9a4f4b49096ca87ed8c7

It turned out, TaskPrototype used `TabStack`, which is still concatenated with the other files, and is not present in the modularized TaskPrototype. I think I'll modularize the TabStack and hopefully the errors will go away, but I think it's good to keep track of them so I'll leave the commit there for some time.

**Comment by code-hunger** [comment]
Also, it's confusing that `TabStack` lives in `tabRestore.js` :) I had to grep the whole source to find it again. Would you mind if I rename the file appropriately?

**Comment by code-hunger** [comment]
Another question: is there a reason `tabPrototype` to be defined this way?

It took me a few minutes to understand this code:

```
    for (var key in tabPrototype) {
      newTask.tabs[key] = tabPrototype[key]
    }
```
Because I expected it to add elements to the `tabs` array, while in fact it adds member functions.

It will be more clear if tabPrototype was defined as 

```
function TabList() {}

TabList.prototype.add = function() {...}
```
and so on. Or even as a ECMA6 class. Then the definition here:

      tabs: task.tabs || [],

will become this:

      tabs: new TabList(task.tabs),

**Comment by PalmerAL** [comment]
> Also, it's confusing that TabStack lives in tabRestore.js :) I had to grep the whole source to find it again. Would you mind if I rename the file appropriately?

Makes sense to me! There's no reason it shouldn't be a module, I just hadn't gotten around to converting it yet.

Converting the list to a TabList object sounds fine as well.

**Comment by code-hunger** [comment]
I moved my other comment just before you responded here, thinking it'd be better to keep the two separated.
